trade-finance-manager-api/src/v1/controllers/update-facilities.js

```diff
@@ -19,11 +19,62 @@ const updateFacilities = async (deal) => {
 
   modifiedDeal.facilities = await Promise.all(modifiedDeal.facilities.map(async (f) => {
     const facility = f;
-    const { _id: facilityId } = facility;
+
+    const {
+      _id: facilityId,
+      hasBeenIssued,
+    } = facility;
+
     let facilityUpdate;
     let facilityPremiumSchedule;
     let feeRecord;
 
+    /**
+     * If facility hasBeenIssued
+     * Check if gef or bss
+     * Adds hasBeenIssuedAndAcknowledged and/or hasBeenAcknowledged parameter
+     * Updates the facility collection with flags and tfm facilities collection
+    */
+    if (hasBeenIssued) {
+      let portalFacilityUpdate;
+
+      if (dealType === CONSTANTS.DEALS.DEAL_TYPE.GEF) {
+        portalFacilityUpdate = {
+          hasBeenIssuedAndAcknowledged: true,
+        };
+        // only changes flag if AIN or MIA
+        if (submissionType !== CONSTANTS.DEALS.SUBMISSION_TYPE.MIA) {
+          // updates GEF facility collection
+          const updatedPortalFacility = await api.updateGefFacility(facilityId, portalFacilityUpdate);
+
+          facility.hasBeenIssuedAndAcknowledged = updatedPortalFacility.hasBeenIssuedAndAcknowledged;
+        }
+      } else {
+        // If BSS/EWCS
+        const facilityStatusUpdate = CONSTANTS.FACILITIES.FACILITY_STATUS_PORTAL.ACKNOWLEDGED;
+
+        await api.updatePortalFacilityStatus(facilityId, facilityStatusUpdate);
+
+        portalFacilityUpdate = {
+          hasBeenIssuedAndAcknowledged: true,
+          hasBeenAcknowledged: true,
+        };
```

Comment:
Could save some repitition of this variable and have `portalFacilityUpdate` at the top like this:

```js
if (hasBeenIssued) {
  const portalFacilityUpdate = {
    hasBeenIssuedAndAcknowledged: true,
  };

  if (dealType === CONSTANTS.DEALS.DEAL_TYPE.GEF) {
    // ...
  }

  if (dealType === CONSTANTS.DEALS.DEAL_TYPE.BSS_EWCS) {
    portalFacilityUpdate.hasBeenAcknowledged = true;
  }
}
```

The BSS specific flag `hasBeenAcknowledged` could be replaced with just `hasBeenIssuedAndAcknowledged`, with find and replace. Not a blocker.

Reply:
portalFacilityUpdate changed to ^